completions/spicetify.lua

```diff
@@ -0,0 +1,177 @@
+--------------------------------------------------------------------------------
+-- Clink argmatcher for Spicetify
+--------------------------------------------------------------------------------
+
+function split(inputstr, sep)
+    if sep == nil then
+        sep = "%s"
+    end
+    local t={}
+    for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
+        table.insert(t, str)
+    end
+    return t
+end
+
+function get_extension_names(append_hyphen)
+    local handle = io.popen("2>nul spicetify.exe path -e")
+    local result = handle:read("*a")
+    handle:close()
+    local paths = split(result, "\n")
+    local names = {}
+    for _, path in ipairs(paths) do
+        local name = path:match("([^\\]+)$")
+        if append_hyphen then
+            name = name .. "-"
+        end
+        table.insert(names, name)
+    end
+    return names
+end
+
+function get_app_names(append_hyphen)
+    local handle = io.popen("2>nul spicetify.exe path -a")
+    local result = handle:read("*a")
+    handle:close()
+    local paths = split(result, "\n")
+    local names = {}
+    for _, path in ipairs(paths) do
+        local name = path:match("([^\\]+)$")
+        if append_hyphen then
+            name = name .. "-"
+        end
+        table.insert(names, name)
+    end
+    return names
+end
+
+local one_arg = clink.argmatcher():addarg()
+
+local function create_arg(name)
+    return clink.argmatcher():addarg(name .. one_arg)
+end
+
+local empty_parser = clink.argmatcher()
+
+local backup_parser = clink.argmatcher()
+    :addarg("apply")
+
+local refresh_parser = clink.argmatcher()
+    :addarg("-e")
+
+local help_parser = clink.argmatcher()
+    :addarg("config")
+
+local watch_parser = clink.argmatcher()
+    :addarg("-e", "-a", "-s", "-l")
+
+local path_a_flag_parser = clink.argmatcher()
+    :addarg("root", get_app_names(false))
```

Comment:
The `get_app_names(false)` here executes `spicetify.exe` when this Lua script is loaded, and then never again.  And the `get_extension_names(false)` executes it a second time when this Lua script is loaded.

It would be better to execute `spicetify.exe` when completion is performed.

To do that, pass only a function name here:
```lua
    :addarg("root", get_app_names)
```

If you want to pass `false` or `true`, then you need an intermediate function:

```lua
local function get_app_names_false()
    return get_app_names(false)
end

local function get_app_names_true()
    return get_app_names(true)
end

local path_a_flag_parser = clink.argmatcher()
    :addarg("root", get_app_names_false)
```

The Lua syntax `function_name` is just a reference to the function, so that the function can get called later, on demand.

The Lua syntax `function_name()` is an immediate call to the function, and then whatever the function returns is used.

It's a subtle detail about efficiency in argmatchers.

Reply:
Ah cool, you learn something new everyday.

Once again, thanks for the explanation.

Just pushed a commit that fixes it.